rust/core-lib/src/editor.rs

```diff
@@ -565,7 +569,8 @@ impl Editor {
 
     fn insert_newline(&mut self) {
         self.this_edit_type = EditType::InsertChars;
-        self.insert("\n");
+        let text = self.config.newline.clone();
```

Comment:
Don't think you need the clone here, can just take a reference.

Reply:
@raphlinus we can't take a reference because we mutably borrow `self` in `self.insert`. I'm not sure if there's a simple workaround for this?

Reply:
Oh, right. Well, the cost of the clone should be trivial in practice. Please proceed.